### PR TITLE
Upgrade net-libs/libmicrohttpd to 0.9.73

### PR DESCRIPTION
**Pull request by ArkaprabhaChakraborty**
This PR should be merged together with https://github.com/flatcar-linux/coreos-overlay/pull/1408.

## Testing done
https://dpaste.org/YyC2

CI passed: http://jenkins.infra.kinvolk.io:8080/job/os/job/manifest/4221/cldsv

### CODE REVIEW

File: net-libs/libmicrohttpd/logs/custom-scanners-service.log [hunk 1]
```diff
@@ -0,0 +1,84 @@
+2021-11-10 02:29:16.084  INFO 286042 --- [main] c.h.a.c.CustomScannersServiceApplication : Starting CustomScannersServiceApplication on fallenvalkyrie with PID 286042 (/home/arkaprabha/.vscode/extensions/hcltechnologies.hclappscancodesweep-1.1.4/lib/custom-scanners-service-1.0.11.jar started by arkaprabha in /home/arkaprabha/flatcar-sdk/src/third_party/portage-stable/net-libs/libmicrohttpd)
```
Comment (dongsupark) [inline net-libs/libmicrohttpd/logs/custom-scanners-service.log:1]: When you move the PR to coreos-overlay, can you please delete `logs/custom-scanners-service.log`?
I suppose, that file is not needed.

Comment (ArkaprabhaChakraborty) [reply net-libs/libmicrohttpd/logs/custom-scanners-service.log:1]: This one isn't supposed to be moved to coreos-overlay. This package is in portage. That was net-libs/rpcsvc-proto.

Comment (ArkaprabhaChakraborty) [reply net-libs/libmicrohttpd/logs/custom-scanners-service.log:1]: But if you want I can remove this file.

**Comment by ArkaprabhaChakraborty** [comment]
@dongsupark removed it. Can you check please check if its correct. This PR is correct I guess. The one to be moved was for net-libs/rpcsvc-proto which I have done already.

**Comment by ArkaprabhaChakraborty** [comment]
> I see, we do not need to move it to coreos-overlay.
> 
> However, can you please rebase this PR on top of the current main branch? At the moment, your portage-stable branch is based on an older tree from Sep. 28th, so building libmicrohttpd is not possible due to a missing eclass.
> 
> ```
> * ERROR: sys-apps/systemd-249.4::coreos failed (depend phase):
>  *   meson-multilib.eclass could not be found by inherit()
> ```
> 
> A quick way to check if the both trees are in sync with current main branches is, to run `emerge-amd64-usr --pretend --verbose net-libs/libmicrohttpd` under the local SDK environment. If it returns a normal output, your trees do not have such a syntax issue. If it returns an error, then you probably need to rebase.

It returned a normal output. The emerge before rebase didn't give any error and it didn't give any error even after I have rebased it.

**Comment by dongsupark** [comment]
Yesterday I simply ran a CI build with this PR and https://github.com/flatcar-linux/coreos-overlay/pull/1408 .

Then the SDK build failed like:
```
emerge: there are no ebuilds built with USE flags to satisfy ">=sys-libs/ncurses-5.9-r3:0=[static-libs?,unicode?,abi_x86_32(-)?,abi_x86_64(-)?,abi_x86_x32(-)?,abi_mips_n32(-)?,abi_mips_n64(-)?,abi_mips_o32(-)?,abi_riscv_lp64d(-)?,abi_riscv_lp64(-)?,abi_s390_32(-)?,abi_s390_64(-)?]".
!!! One of the following packages is required to complete your request:
- sys-libs/ncurses-6.2_p20210619::coreos (Missing IUSE: unicode)
```

**Comment by ArkaprabhaChakraborty** [comment]
sys-libs/ncurses is 6.2 in coreos but libmicrohttd supports upto ncurses 5.9 ( I guess that's what the error says ).

**Comment by ArkaprabhaChakraborty** [comment]
So this is a package bug?

**Comment by dongsupark** [comment]
The error above happens because ncurses >= 6.2_p20210619 now does not have the `unicode` USE flag in its IUSE. 
That's why we had to update several ebuilds in portage-stable, https://github.com/flatcar-linux/portage-stable/pull/236 , so they can make the `unicode` flag optional.
See also https://github.com/flatcar-linux/coreos-overlay/pull/1383.
As your portage-stable branch is still based on [an older version](https://github.com/flatcar-linux/portage-stable/pull/243#issuecomment-965338075) of main branch, Gentoo portage still tries to pull in ncurses that still requires the `unicode` flag, not optional.
That's one of the common issues that we encounter when dealing with multiple Gentoo overlay trees that are not exactly in-sync.
That's why I asked you to rebase your tree on top of the current main branch.

The warnings we saw around systemd and meson-multilib above are just warnings, so it might not be necessarily a blocker.
However, that's not the only issue there.
Basically if your trees are not in sync, you still could bump into such strange issues like the ncurses one.

**Comment by ArkaprabhaChakraborty** [comment]
@dongsupark Can you check and tell me now? I have made the changes on the branch from a newly installed sdk.

**Comment by ArkaprabhaChakraborty** [comment]
emerge-amd64-usr net-libs/libmicrohttpd  gives
```
>>> Installing (1 of 1) net-libs/libmicrohttpd-0.9.73::portage-stable to /build/amd64-usr/
 * Running stacked hooks for pre_pkg_preinst
 *    wrap_old_config_scripts ...                                                                                                                                     [ ok ]
>>> Auto-cleaning packages...
```

And emerge-amd64-usr --pretend --verbose net-libs/libmicrohttpd gives
```
These are the packages that would be merged, in order:

Calculating dependencies... done!
[ebuild   R    ] net-libs/libmicrohttpd-0.9.73:0/12::portage-stable to /build/amd64-usr/ USE="epoll -ssl -static-libs -test -thread-names" 0 KiB

Total: 1 package (1 reinstall), Size of downloads: 0 KiB

```

**Comment by dongsupark** [comment]
Thanks, it looks better now.

CI running: http://jenkins.infra.kinvolk.io:8080/job/os/job/manifest/4221/cldsv

**Comment by dongsupark** [comment]
CI passed.
So there is no significant regression.
Before approving the PR, let me check if we need to update other packages that depend on libmicrohttpd.

**Comment by ArkaprabhaChakraborty** [comment]
> CI passed. So there is no significant regression. Before approving the PR, let me check if we need to update other packages that depend on libmicrohttpd.

If we need to upgrade some packages, can I work on them ?

**Review by dongsupark — APPROVED** [review]
I have gone through comments of https://bugs.gentoo.org/779151, to check any ebuilds there is included in Flatcar.
Apparently no ebuild of them is in Flatcar.
So we do not have to worry about that.

Thanks for the contribution!
